elliottlib/cli/__main__.py

```diff
@@ -200,12 +208,37 @@ def find_bugs(runtime, advisory, default_advisory_type, mode, status, id, from_d
                                                                                     bug.summary[:60]))
 
     if len(flag) > 0:
-        for bug in bug_ids:
+        for bug in bugs:
             for f in flag:
                 bug.updateflags({f: "+"})
 
-    if advisory is not False:
-        elliottlib.errata.add_bugs_with_retry(advisory, [bug.id for bug in bug_ids], False)
+    if advisory is not False and not default_advisory_type:  # `--add ADVISORY_NUMBER` should respect the user's wish and attach all available bugs to whatever advisory is specified.
+        elliottlib.errata.add_bugs_with_retry(advisory, [bug.id for bug in bugs], False)
+        return
+
+    # if --use-default-advisory or --into-default-advisories is given, we need to determine which bugs should be swept into which advisory
```

Comment:
and if not, we don't need to sweep bugs at all... that may be worth saying here. or simply move this comment below the conditional.

Reply:
I've amended the comment.